config/m4/compiler.m4

```diff
@@ -17,8 +17,19 @@ AC_ARG_ENABLE(debug,
         [],
         [enable_debug=no])
 AS_IF([test "x$enable_debug" == xyes],
-        [BASE_CFLAGS="-O0 -D_DEBUG $BASE_CFLAGS"],
-        [BASE_CFLAGS="-O3 $BASE_CFLAGS"])
+        [BASE_CFLAGS="-D_DEBUG $BASE_CFLAGS"],
+        [])
+
+#
+# Optimization level
+#
+AC_ARG_ENABLE(fast,
+        AC_HELP_STRING([--enable-fast], [Set optimization level [0-3]]),
+        [],
+        [enable_fast=""])
+AS_IF([test -z "$enable_fast"], [],
+      [test "$enable_fast" == "yes"], [BASE_CFLAGS="-O3 $BASE_CFLAGS"],
+      [BASE_CFLAGS="-O$enable_fast $BASE_CFLAGS"])
```

Comment:
now the default behave has changed:

before this PR - O3 was enable by default
after this PR - no optimization

maybe we should have enable_fast=yes by default to match default behavior?

Reply:
ok, updated...